spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/PubSubHeaderMapper.java

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.pubsub.integration;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.springframework.integration.mapping.HeaderMapper;
+import org.springframework.integration.util.PatternMatchUtils;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.util.Assert;
+
+/**
+ * Maps headers from {@link com.google.pubsub.v1.PubsubMessage}s to
+ * {@link org.springframework.messaging.Message}s and vice-versa.
+ *
+ * <p>By default, filters out headers called "id" or "timestamp" on the
+ * {@link org.springframework.messaging.Message} to {@link com.google.pubsub.v1.PubsubMessage}
+ * header conversion.
+ *
+ * @author João André Martins
+ */
+public class PubSubHeaderMapper implements HeaderMapper<Map<String, String>> {
+
+	/**
+	 * Patterns of headers to map in {@link #fromHeaders(MessageHeaders, Map)}.
+	 */
+	private String[] outboundHeaderPatterns = {
+			"!" + MessageHeaders.ID,
+			"!" + MessageHeaders.TIMESTAMP,
```

Comment:
We need to add to this set to exclude also a `GcpPubSubHeaders.ACKNOWLEDGEMENT` header as well.
This one is populated by the `PubSubInboundChannelAdapter` and should not be sent to the Pub/Sub back.